db/queries/general.sql

```diff
@@ -0,0 +1,58 @@
+-- name: GetShopInfo :one
+
+SELECT
+    "seller_name",
+    "image_id",
+    "name",
+    "description"
+FROM "shop"
+WHERE "seller_name" = $1;
+
+-- name: ShopExists :one
+
+SELECT "id"
+FROM "shop" AS s
+WHERE
+    s."seller_name" = $1
+    AND s."enabled" = TRUE;
+
+-- name: GetShopCoupons :many
+
+SELECT
+    "id",
+    "type",
+    "scope",
+    "name",
+    "description",
+    "discount",
+    "start_date",
+    "expire_date"
+FROM "coupon"
+WHERE "shop_id" = $1
+ORDER BY "id"
+LIMIT $2
+OFFSET $3;
+
+-- name: GetTagInfo :one
+
+SELECT "id", "name" FROM "tag" WHERE "id" = $1;
+
+-- name: GetProductInfo :one
+
+SELECT
+    "id",
+    "version",
+    "name",
+    "description",
+    "price",
+    "image_id",
+    "exp_date",
+    "stock",
+    "sales",
+    "enabled"
+FROM "product"
+WHERE "id" = $1;
```

Comment:
check the product is enabled and I dont know we should return version of product or not